src/Microsoft.AspNet.Mvc.TagHelpers/ValidationMessageTagHelper.cs

```diff
@@ -53,14 +53,14 @@ public class ValidationMessageTagHelper : TagHelper
                     {
                         var childContent = await context.GetChildContentAsync();
 
-                        if (string.IsNullOrWhiteSpace(childContent))
+                        if (childContent.IsWhiteSpace)
                         {
                             // Provide default label text since there was nothing useful in the Razor source.
-                            output.Content = tagBuilder.InnerHtml;
+                            output.Content.Append(tagBuilder.InnerHtml);
                         }
                         else
                         {
-                            output.Content = childContent;
+                            childContent.CopyTo(output.Content);
```

Comment:
Depending on the result of this conversation: https://github.com/aspnet/Razor/pull/312/files#r25555141

This may need to go to `Clear` => `CopyTo`